main/cache/cache.cc

```diff
@@ -115,7 +115,7 @@ bool cacheTreesAndFiles(const core::GlobalState &gs, WorkerPool &workers, absl::
                     }
 
                     auto &file = job->file.data(gs);
-                    if (!job->cached() && !file.hasIndexErrors()) {
+                    if (!file.hasIndexErrors()) {
```

Comment:
This is not a great change, I think. As we can't tell if the tree originated in the cache outside of the indexer, we're forced to always write to the cache if it didn't have indexer errors.

Reply:
This could be quite expensive because it means that we will hash the full file contents.

Reply:
Yep, I think this kills this PR for me :(